src/NonlinearArithmetic/Internals/DivInternals.dfy

```diff
@@ -0,0 +1,196 @@
+// RUN: %dafny /compile:0 /noNLarith "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+/***********************************************************************************
+*  Original: Copyright (c) Microsoft Corporation
+*  SPDX-License-Identifier: MIT
+*  
+*  Modifications and Extensions: Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT 
+************************************************************************************/
+
+/* lemmas and functions in this file are used in the proofs in DivMod.dfy 
+
+Specs/implements mathematical div and mod, not the C version.
+This may produce "surprising" results for negative values
+For example, -3 div 5 is -1 and -3 mod 5 is 2.
+Note this is consistent: -3 * -1 + 2 == 5 */
```

Comment:
I find this comment strange because I don't know what C's div and mod do (these results aren't surprising to me). Maybe you could mention the consistency property you're alluding to? I think it's `(x div n) * n + (x mod n) == x`, and maybe it would help to mention `0 <= x mod n < n`.